airflow/utils/context.py

```diff
@@ -260,14 +212,13 @@ def __getitem__(self, key: int | Asset | AssetAlias | AssetRef) -> LazyAssetEven
                 raise KeyError(obj) from None
             join_clause = AssetEvent.asset
             where_clause = and_(AssetModel.uri == asset.uri, AssetModel.active.has())
-        else:
-            raise ValueError(key)
 
-        return LazyAssetEventSelectSequence.from_select(
-            select(AssetEvent).join(join_clause).where(where_clause),
-            order_by=[AssetEvent.timestamp],
-            session=self._session,
-        )
+        with create_session() as session:
+            return LazyAssetEventSelectSequence.from_select(
+                select(AssetEvent).join(join_clause).where(where_clause),
+                order_by=[AssetEvent.timestamp],
+                session=session,
+            )
```

Comment:
When do we actually need this implementation?

Reply:
It's now removed